[Documentation] Add How to configure an Azure Application Gateway (#463)
southworks/botframework-solutions: docs/_docs/virtual-assistant/handbook/configure-an-azure-application-gateway.md

## 1. Intro

This explains how to configure and use a bot through the public IP of an Azure Application Gateway.

> 

Reviewer: ```suggestion
---
category: Virtual Assistant
subcategory: Handbook
title: Configure an Azure Application Gateway
description: This explains how to configure and use a bot through the public IP of an Azure Application Gateway.
order: 14
toc: true
---

# {{ page.title }}
{:.no_toc}
{{ page.description }}

> ### Prerequisites

Reviewer: ```suggestion
## Prerequisites
```

Author: Should it be the prerequisites be at the same levels of the other sections?
I see it too small, maybe either expand it or incorporate it into the intro

Reviewer: Compare it with other documents that contains prerequisites!

> 1.  Have a deployed [Virtual Assistant](https://microsoft.github.io/botframework-solutions/overview/virtual-assistant-solution/)
> 2.  The [BotFramework Emulator](https://github.com/microsoft/BotFramework-Emulator)

Reviewer: To have consistency with other docs:

```suggestion
-  Have a deployed [Virtual Assistant]({{site.baseurl}}/overview/virtual-assistant-solution/)
-  The [BotFramework Emulator](https://github.com/microsoft/BotFramework-Emulator)
```

> ## 2. Create an Azure Application Gateway

Reviewer: ```suggestion
## Create an Azure Application Gateway
```

Follow the document [Tutorial: Create an application gateway with a Web Application Firewall using the Azure portal](https://docs.microsoft.com/en-us/azure/web-application-firewall/ag/application-gateway-web-application-firewall-portal) up to *Review + create tab* section.

You should end up with the following resources in your Azure resource group:

> ![]({{site.baseurl}}/assets/images/virtualassistant-gateway-resources.png)

Reviewer: What can we add inside `[]`? Is it the title of the image right? Can we fill it?

Author: Can or should we? I used as example an image used in `activity-flow.md` which has the same format

Reviewer: Ack

> ## 3. Configure the Gateway's backend pool

Reviewer: ```suggestion
## Configure the Gateway's backend pool
```

Now it is necessary to link your deployed bot to the backend pool of your application gateway. For this follow the steps detailed in [Add App service as backend pool](https://docs.microsoft.com/en-us/azure/application-gateway/configure-web-app-portal#add-app-service-as-backend-pool) and in [Edit HTTP settings for App Service](https://docs.microsoft.com/en-us/azure/application-gateway/configure-web-app-portal#edit-http-settings-for-app-service).

> **Note**: If you are using a 1.0 TypeScript Virtual Assistant or Skill, check this known issue: *How to configure a TypeScript bot with an Application Gateway*.

Reviewer: Link to the know issue?

After overriding the hostname, you can use the gateway's public IP to access your bot.

![]({{site.baseurl}}/assets/images/virtualassistant-gateway-emulator.png)

> ## 4. Further Reading

Reviewer: ```suggestion
## Further Reading
```

> - *How to configure a TypeScript bot with an Application Gateway*

Reviewer: Is this okay?

Author: Not at all, this must be deleted

- [How to configure a TypeScript bot with an Application Gateway]({{site.baseurl}}/help/known-issues#configuring-a-gateway's-health-prob- with-a-typeScript-bot)
- [Deploying a Virtual Assistant](https://microsoft.github.io/botframework-solutions/virtual-assistant/tutorials/deploy-assistant/cli/1-intro/)
- [Troubleshoot backend health issues in Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-backend-health-troubleshooting)

> - [Troubleshooting bad gateway errors in Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-troubleshooting-502)

Reviewer: Is it okay the order?

Author: I will switch the first and the second

{:.no_toc}
{{ page.description }}

### Prerequisites

-  Have a deployed [Virtual Assistant]({{site.baseurl}}/overview/virtual-assistant-solution/)
-  The [BotFramework Emulator](https://github.com/microsoft/BotFramework-Emulator)

## Create an Azure Application Gateway

Follow the document [Tutorial: Create an application gateway with a Web Application Firewall using the Azure portal](https://docs.microsoft.com/en-us/azure/web-application-firewall/ag/application-gateway-web-application-firewall-portal) up to *Review + create tab* section.

You should end up with the following resources in your Azure resource group:

![]({{site.baseurl}}/assets/images/virtualassistant-gateway-resources.png)

## Configure the Gateway's backend pool

Now it is necessary to link your deployed bot to the backend pool of your application gateway. For this follow the steps detailed in [Add App service as backend pool](https://docs.microsoft.com/en-us/azure/application-gateway/configure-web-app-portal#add-app-service-as-backend-pool) and in [Edit HTTP settings for App Service](https://docs.microsoft.com/en-us/azure/application-gateway/configure-web-app-portal#edit-http-settings-for-app-service).

> **Note**: If you are using a TypeScript bot sample, check [How to configure a TypeScript bot with an Application Gateway]({{site.baseurl}}/help/known-issues#configuring-a-gateway's-health-prob- with-a-typeScript-bot).

Reviewer: ```suggestion
> If you are using a TypeScript bot sample, check [How to configure a TypeScript bot with an Application Gateway]({{site.baseurl}}/help/known-issues#configuring-a-gateway's-health-probe-with-a-typescript-bot).
```

After overriding the hostname, you can use the gateway's public IP to access your bot.

![]({{site.baseurl}}/assets/images/virtualassistant-gateway-emulator.png)

## Further Reading

- [Deploying a Virtual Assistant](https://microsoft.github.io/botframework-solutions/virtual-assistant/tutorials/deploy-assistant/cli/1-intro/)
- [How to configure a TypeScript bot with an Application Gateway]({{site.baseurl}}/help/known-issues#configuring-a-gateway's-health-prob- with-a-typeScript-bot)
- [Troubleshoot backend health issues in Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-backend-health-troubleshooting)

> - [Troubleshooting bad gateway errors in Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-troubleshooting-502)

Reviewer: The second document (`How to configure a TypeScript bot with an Application Gateway`) should be removed as it's already pointed above
